src/main/java/org/mockbukkit/mockbukkit/inventory/meta/FireworkMetaMock.java

```diff
@@ -35,11 +36,14 @@ public FireworkMetaMock()
 	 *
 	 * @param meta The meta to clone.
 	 */
-	public FireworkMetaMock(@NotNull FireworkMeta meta)
+	public FireworkMetaMock(@NotNull ItemMeta meta)
 	{
 		super(meta);
 
-		this.effects.addAll(meta.getEffects());
+		if(meta instanceof FireworkMeta fireworkMeta)
+		{
+			this.effects.addAll(fireworkMeta.getEffects());
```

Comment:
This is not copying the power of the firework, is this intended?

Reply:
It was not intended thanks !